### PR TITLE
Add multithreaded build

### LINKED ISSUE #13: Build Feature - Threading
A must. Already working on it.

`build -j --jobs <jobs>`

### PR DESCRIPTION
**Pull request by bovine3dom**
Uses rayon to multithread building of mods and optionals by folder.

Accepts `-j` and `--jobs` flags to change at runtime. Defaults to number
of logical processors.

Authors: bovine3dom and jonpas. Squashed for clarity.

NB: untested on Windows. Works fine on Linux.

Closes #13.

### CODE REVIEW

**Comment by bovine3dom** [comment]
Force pushed a couple of times to fix a few errors that had slipped past me in the rebase. I'll stop now :)

File: src/build.rs [hunk 1]
```diff
@@ -12,6 +12,8 @@ use std::time::{Duration, SystemTime};
 use crate::error;
 use crate::error::*;
 
+use rayon::prelude::*;
```
Comment (jonpas) [inline src/build.rs]: Move above to other external crates.

File: src/build.rs [hunk 2]
```diff
@@ -50,12 +55,12 @@ pub fn build_single(p: &crate::project::Project, addon: &String) -> Result<(), E
...
 }
 
-pub fn release(p: &crate::project::Project, version: &String) -> Result<(), Error> {
+pub fn release(p: &crate::project::Project, version: &String, jobs: &usize) -> Result<(), Error> {
     println!(" {} release v{}", "Preparing".green().bold(), version);
     if Path::new(&format!("releases/{}", version)).exists() {
         return Err(error!("Release already exists, run with --force to clean"));
     }
-    build(&p)?;
+    build(&p, jobs)?;
```
Comment (jonpas) [inline src/build.rs]: Borrow? &

**Comment by bovine3dom** [comment]
Just tested on Windows. Works great.

Build times have gone from about 10-20 minutes with Mikero tools (and not being able to use the PC in that time because it does so much focus stealing) to about a minute or two with HEMTT. ❤️

File: Cargo.toml [hunk 3]
```diff
@@ -15,6 +15,8 @@ serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 serde-xml-rs = "0.3.1"
 walkdir = "2.2"
+rayon = "1.0"
```
Comment (synixebrett) [inline Cargo.toml]: All the others are in alphabetical order

File: src/build.rs [hunk 4]
```diff
@@ -27,7 +27,7 @@ pub fn modtime(addon: &Path) -> Result<SystemTime, Error> {
 }
 
 pub fn build(p: &crate::project::Project, jobs: &usize) -> Result<(), Error> {
-    rayon::ThreadPoolBuilder::new().num_threads(*jobs).build_global().unwrap();
+    Some(rayon::ThreadPoolBuilder::new().num_threads(*jobs).build_global());
```
Comment (jonpas) [inline src/build.rs]: Add a comment as to why (as you did in the commit message)?

File: src/main.rs [hunk 5]
```diff
@@ -41,7 +42,7 @@ Usage:
     hemtt init
     hemtt create
     hemtt addon <name>
-    hemtt build [<addons>] [--release] [--force] [--nowarn] [--opts=<addons>] [--skip=<addons>]
+    hemtt build [<addons>] [--release] [--force] [--nowarn] [--opts=<addons>] [--skip=<addons>] [---jobs=<n>]
```
Comment (jonpas) [inline src/main.rs]:
```suggestion
    hemtt build [<addons>] [--release] [--force] [--nowarn] [--opts=<addons>] [--skip=<addons>] [--jobs=<n>]
```

Comment (jonpas) [reply src/main.rs]: @synixebrett should we start putting short flags where they exist in this line to make it shorter? They are explained below and most of them are the same as any CLI tool.

Comment (synixebrett) [reply src/main.rs]: Probably should yeah, it's only going to get longer.

**Comment by bovine3dom** [comment]
I multithreaded all the things :)

**Pull request merged**